llvm/lib/Target/DirectX/DXILDataScalarization.cpp

```diff
@@ -83,6 +91,42 @@ DataScalarizerVisitor::lookupReplacementGlobal(Value *CurrOperand) {
   return nullptr; // Not found
 }
 
+// Recursively Creates and Array like version of the given vector like type.
```

Comment:
nit: I know the original comment read like this but I think it may as well be changed now to make more sense.
```suggestion
// Recursively creates an array-like version of the given vector type.
```